src/DDTrace/OpenTelemetry/SpanBuilder.php

```diff
@@ -92,6 +96,38 @@ public function addLink(SpanContextInterface $context, iterable $attributes = []
         return $this;
     }
 
+    public function addEvent(string $name, int $timestamp = null, iterable $attributes = []): SpanBuilderInterface
+    {
+        $attributesArray = Attributes::create($attributes);
+        $nanoTimestamp = ($timestamp !== null && $timestamp < 1e9) ? $timestamp * 1000 : ($timestamp ?? (int)(microtime(true) * 1e9)); // Convert microseconds to nanoseconds if needed
+        $this->events[] = new Event($name, $nanoTimestamp, Attributes::create($attributes));
+        return $this;
+    }
+
+    public function recordException(Throwable $exception, iterable $attributes = []): SpanBuilderInterface
+    {
+        // Standardized exception attributes
+        $exceptionAttributes = [
+            'exception.message' => $attributes['exception.message'] ?? $exception->getMessage(),
+            'exception.type' => $attributes['exception.type'] ?? get_class($exception),
+            'exception.stacktrace' => $attributes['exception.stacktrace'] ?? $exception->getTraceAsString(),
+            'exception.escaped' => false
+        ];
+
+        // Update span metadata based on exception attributes
+        $this->setAttribute(Tag::ERROR_MSG, $exceptionAttributes['exception.message']);
+        $this->setAttribute(Tag::ERROR_TYPE, $exceptionAttributes['exception.type']);
+        $this->setAttribute(Tag::ERROR_STACK, $exceptionAttributes['exception.stacktrace']);
```

Comment:
Are you sure about this? Exceptions generally should only be added on spans if they escape their span.

Reply:
This is by design, to match what our Datadog Agent does for OTEL Events when it receives OTLP traces. If there are any `exception` events on a span, then the last event that is added will be used to populate the error tagging fields on the span.

Note: We do not update the `span.Error` field when calling this method. To update the `span.Error` field, the user would have to call the OpenTelemetry `span.SetStatus` API with the Error status.

Reply:
`error` is set automatically when `error.message` or `error.type` are present. So, at least that assumption is not correct here.

When manually adding errors, users are supposed to only add `error.message` and `error = 1` is automatically added for them.

Reply:
Also IIRC that behaviour of populating error.message is a fallback behaviour from the ages where no span event support existed in the UI.

Reply:
Hey @bwoebi, I've been looking into this ask here since I thought for us to work around this I could check if the span was an error or not, if not then reset the `error` tag but I am not sure how to accomplish this, for example I saw that the error can be removed from the `metrics` object using ` unset($this->span->metrics[Tag::ERROR]);` but couldn't find a way to do so for the body of the Span and wanted to check what you'd recommend.

Reply:
What I would recommend is not doing this, honestly.
Like, it's a bad idea which goes against the defined semantics of the tracer.

Reply:
I see what you mean, will go ahead with only adding the `stack` tag for now and later connect with Error Tracking to  try getting the details from the span event instead, does that sounds better at the moment?

Reply:
Yes.